packages/sanity/src/core/templates/types.ts

```diff
@@ -25,58 +83,117 @@ export interface TemplateReferenceTarget {
+   */
   name: string
+
+  /**
+   * Parameter type, eg `string`, `number`, `boolean` etc.
+   */
   type: string
+
+  /**
+   * Parameter type. Will be attempted to be automatically set if not given,
+   * by title-casing the `name` property.
+   */
   title?: string
+
+  /**
+   * Description for the parameter. Optional.
+   * May be used in the future to explain the parameter in UIs.
+   */
   description?: string
+
+  /**
+   * Optional bag of options for the parameter. Currently unused.
+   */
   options?: {[key: string]: any}
 }
 
-/** @internal */
-export type ReferenceFieldDefinition = TemplateFieldDefinition & TemplateReferenceTarget
-
 /** @public */
 export type TemplateArrayFieldDefinition = TemplateFieldDefinition & {
   type: 'array'
   of: (TemplateReferenceTarget | TypeTarget)[]
 }
 
 /**
- * Interface for initial value template item
+ * Representation of an initial value template _item_
+ * Used by the {@link desk.StructureBuilder} class to determine which initial value templates
+ * should be available for a desk structure node, such as a list pane.
  *
  * @public
  */
 export interface InitialValueTemplateItem extends TemplateItem {
   type: 'initialValueTemplateItem'
-  /** Initial value schema type */
+
+  /** ID for this template item */
+  id: string
+
+  /** Initial value template schema type */
   schemaType: string
 }
 
 /**
- * Interface for template response
+ * Represents the items that can appear in different parts of the Sanity studio when creating
+ * new documents - examples being the "New document" button in the navigation bar,
+ * the corresponding button in panes, as well as the "Create new" button on references.
+ *
+ * Differs from an actual _template_ in that a single template can be pointed at by multiple
+ * different items. This is useful when the template can create different values based on
+ * passed parameters.
  *
  * @public
  */
 export interface TemplateItem {
   /**
-   * ID for the template the item references
+   * ID for the template. Must be unique within the set of templates.
    */
   templateId: string
-  /** Template title */
+
+  /**
+   * Title for the item.
+   * Defaults to the title of the associated template.
+   */
   title?: string
-  /** template subtitle */
-  subtitle?: string
-  /** template description */
-  description?: string
-  /** template parameters */
+
+  /**
+   * Parameters for the template - an object of any JSON-serializable values
+   */
   parameters?: {[key: string]: any}
-  /** template icon */
-  icon?: React.ElementType | React.ReactElement
+
+  /**
+   * React icon for the item, if any.
+   * Defaults to the icon for the associated template.
+   */
+  icon?: ElementType | ReactElement
+
   /**
-   * The id of the document to use as initial value
+   * Experimental: not fully supported yet
+   * Hints at what the document ID for the new document should be.
+   * Leave undefined to let the system decide.
+   *
    * @experimental
+   * @beta
```

Comment:
@rexxars should we add @hidden on this as well? Or are we ok with keeping it beta & experimental?